gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQuery.java

```diff
@@ -25,5 +34,602 @@
  */
 public interface BigQuery extends Service<BigQueryOptions> {
 
-  // TODO(mziccard) add missing methods
+  enum DatasetField {
+    ACCESS("access"),
+    CREATION_TIME("creationTime"),
+    DATASET_REFERENCE("datasetReference"),
+    DEFAULT_TABLE_EXPIRATION_MS("defaultTableLifetime"),
+    DESCRIPTION("description"),
+    ETAG("etag"),
+    FRIENDLY_NAME("friendlyName"),
+    ID("id"),
+    LAST_MODIFIED_TIME("lastModifiedTime"),
+    LOCATION("location"),
+    SELF_LINK("selfLink");
+
+    private final String selector;
+
+    DatasetField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(DatasetField... fields) {
+      HashSet<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
```

Comment:
Just thinking out loud. If enum fields implemented a "Field" interface with a `String selector()` method then you can have only one static selector method `static String selector(Selector first, Selector... other)` for all...